src/arrow/rle_decoder.h

```diff
@@ -0,0 +1,338 @@
+      int repeat_batch =
+          std::min(batch_size - values_read, static_cast<int>(repeat_count_));
+      std::fill(values + values_read, values + values_read + repeat_batch,
+                static_cast<T>(current_value_));
+      repeat_count_ -= repeat_batch;
+      values_read += repeat_batch;
+    } else if (literal_count_ > 0) {
+      int literal_batch =
+          std::min(batch_size - values_read, static_cast<int>(literal_count_));
+      int actual_read =
+          bit_reader_.GetBatch(bit_width_, values + values_read, literal_batch);
+      DCHECK_EQ(actual_read, literal_batch);
+      literal_count_ -= literal_batch;
+      values_read += literal_batch;
+    } else {
+      if (!NextCounts<T>()) return values_read;
+    }
+  }
+
+  return values_read;
+}
+
+
+template <typename T>
+inline int RleDecoder::GetBatchWithDict(const T* dictionary, T* values, int batch_size) {
+  DCHECK_GE(bit_width_, 0);
+  int values_read = 0;
+  
+  std::vector<uint16_t> isRleVector;
+  std::vector<uint32_t> rleRuns; // esto seria los repeat_count_ para cuando es RLE y literal_coun$
+  std::vector<uint64_t> rleValues; // esto seria los current_value_ para cuando es RLE y 0 para cu$
+  int numRle;
+  int numBitpacked;
+  std::vector<int> unpack32InputOffsets, unpack32OutputOffsets;
+  std::vector<int> remainderInputOffsets, remainderBitOffsets, remainderSetSize, remainderOutputOffsets;
+
+
+
+
+  while (values_read < batch_size) {
+    if (repeat_count_ > 0) {
+      int repeat_batch =
+          std::min(batch_size - values_read, static_cast<int>(repeat_count_));
+  
+        // std::cout<<"repeat_batch: "<<repeat_batch<<std::endl;
+        rleRuns.push_back(repeat_batch);
+        isRleVector.push_back(1);
+        rleValues.push_back(current_value_);
+        numRle++;
+        
+      std::fill(values + values_read, values + values_read + repeat_batch,
+                dictionary[current_value_]);
+      repeat_count_ -= repeat_batch;
+      values_read += repeat_batch;
+    } else if (literal_count_ > 0) {
+      int literal_batch =
+          std::min(batch_size - values_read, static_cast<int>(literal_count_));
+
+      const int buffer_size = 1024;
+      int indices[buffer_size];
+      literal_batch = std::min(literal_batch, buffer_size);
+      
+      // std::cout<<"literal_batch: "<<literal_batch<<std::endl;
+      rleRuns.push_back(literal_batch);
+      isRleVector.push_back(0);
+      rleValues.push_back(0);
+      numBitpacked++;
+
+      bit_reader_.SetGpuBatchMetadata(bit_width_, literal_batch, values_read,
+                                      unpack32InputOffsets, unpack32OutputOffsets,
+                                      remainderInputOffsets, remainderBitOffsets, remainderSetSize, remainderOutputOffsets);
+
+      // std::cout << "values_read:" << values_read << " \n";
+      // std::cout << "bit_width_:" << bit_width_ << " \n";
+      // std::cout << "literal_batch:" << literal_batch << " \n";
```

Comment:
These comments should probably be removed.

Reply:
Thanks. All comments into the api code were removed.